sqlglot/dialects/tsql.py

```diff
@@ -661,8 +661,13 @@ def create_sql(self, expression: exp.Create) -> str:
             exists = expression.args.pop("exists", None)
             sql = super().create_sql(expression)
 
+            table = expression.find(exp.Table)
+
+            if kind == "TABLE" and expression.expression:
+                assert table
+                sql = f"SELECT * INTO {exp.table_name(table)} FROM ({self.sql(expression.expression)}) AS temp"
```

Comment:
why are you using table_name? what if it needs quotes? why not self.sql(table) ?

Reply:
When testing, MS-SQL errored if quotes were used for the table name no matter what type of escaping I used. After further digging, it appears that there is an MS-SQL setting to control use of quotes table names that had been set to off in my testing instance (not sure how, it is on by default).

Will fix.